packages/core/tracing/src/components/trace-viewer/TraceViewer.vue

```diff
@@ -0,0 +1,221 @@
+      size="50"
+    >
+      <div v-if="selectedSpan && !spanNothingToDisplay">
+        <SpanEventList
+          v-if="selectedSpan.span.events && selectedSpan.span.events.length > 0"
+          :span="selectedSpan.span"
+        />
+        <SpanAttributeTable
+          v-if="selectedSpan.span.attributes.length > 0"
+          :span="selectedSpan.span"
+        />
+      </div>
+      <div
+        v-else
+        class="empty-state"
+      >
+        <template v-if="!selectedSpan">
+          {{ t('trace_viewer.empty_state.unselected') }}
+        </template>
+        <template v-else>
+          {{ t('trace_viewer.empty_state.nothing_to_display') }}
+        </template>
+      </div>
+    </Pane>
+  </Splitpanes>
+</template>
+
+<script setup lang="ts">
+import { Pane, Splitpanes } from '@kong/splitpanes'
+import { computed, provide, reactive, shallowRef, type PropType } from 'vue'
+import composables from '../../composables'
+import { TRACE_VIEWER_CONFIG } from '../../constants'
+import type { SpanNode, TraceViewerConfig } from '../../types'
+import WaterfallLegend from '../waterfall/WaterfallLegend.vue'
+import WaterfallView from '../waterfall/WaterfallView.vue'
+import SpanAttributeTable from './SpanAttributeTable.vue'
+import SpanEventList from './SpanEventList.vue'
+
+import '@kong/splitpanes/dist/splitpanes.css'
+
+const { i18n: { t } } = composables.useI18n()
+
+const props = defineProps({
+  config: {
+    type: Object as PropType<TraceViewerConfig>,
+    required: true,
+  },
+  rootSpan: {
+    type: Object as PropType<SpanNode>,
+    required: true,
+  },
+  url: {
+    type: String,
+    required: true,
+  },
+})
+
+// Provide the config to all children components
+provide<TraceViewerConfig | undefined>(TRACE_VIEWER_CONFIG, reactive(props.config))
+
+const selectedSpan = shallowRef<SpanNode | undefined>(undefined)
+
+const handleUpdateSelectedSpan = (span?: SpanNode) => {
+  selectedSpan.value = span
+}
+
+const spanNothingToDisplay = computed(() => {
+  if (!selectedSpan.value) {
+    return true
+  }
+
+  if (selectedSpan.value.span.events && selectedSpan.value.span.events.length > 0) {
+    return false
+  }
+
+  if (selectedSpan.value.span.attributes.length > 0) {
+    return false
+  }
+
+  return true
+})
+</script>
+
+<style lang="scss" scoped>
+.trace-viewer {
+  box-sizing: border-box;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  overflow: hidden;
+  padding: 0 !important;
+  position: relative;
+  width: 100%;
+
+  :deep(.splitpanes__splitter) {
+    $resize-handle-height: 4px;
+    align-items: center;
+    border-top: 1px solid $kui-color-border-neutral-weaker;
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    padding: $kui-space-20 0;
+    position: relative;
+    width: 100%;
+
+    &::before {
+      background-color: $kui-color-background-neutral-weaker;
+      content: '';
+      height: 2px;
+      position: absolute;
+      top: -1.5px;
```

Comment:
How is this one calculated?

Reply:
I drew an ugly chart and hope it would be helpful for understanding:

<img width="1222" alt="Screenshot 2024-11-21 at 17 31 11" src="https://github.com/user-attachments/assets/0a307764-4c58-4786-b88a-36c3f1a97586">

Reply:
Yeah I mean maybe we could use something like `-0.5 * ($border-width + $animated-border-width)`